feat(hashicorp): Add `VAULT_TOKEN` support

V0lantis opened the pull request:
`VAULT_TOKEN` wasn't checked upon `VaultClient`construction. Added a check to ingest it if present in os.environ. See #37336 

<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->



<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

V0lantis committed:
feat(hashicorp): Add `VAULT_TOKEN` support

`VAULT_TOKEN` wasn't checked upon `VaultClient`
construction. Added a check to ingest it if
present in os.environ

Taragolis commented on tests/providers/hashicorp/_internal_client/test_vault_client.py:
```diff
@@ -551,6 +552,19 @@ def test_token(self, mock_hvac):
         assert "secret" == vault_client.mount_point
 
+    @mock.patch("airflow.providers.hashicorp._internal_client.vault_client.hvac")
+    def test_token_in_env(self, mock_hvac):
```
```suggestion
    def test_token_in_env(self, mock_hvac, monkeypatch):
        monkeypatch.setenv("VAULT_TOKEN", "s.7AU0I51yv1Q1lxOIg1F3ZRAS")
```

Taragolis commented on tests/providers/hashicorp/_internal_client/test_vault_client.py:
```diff
@@ -551,6 +552,19 @@ def test_token(self, mock_hvac):
+    def test_token_in_env(self, mock_hvac):
+        mock_client = mock.MagicMock()
+        mock_hvac.Client.return_value = mock_client
+        os.environ["VAULT_TOKEN"] = "s.7AU0I51yv1Q1lxOIg1F3ZRAS"
```
```suggestion
```

Taragolis commented on tests/providers/hashicorp/_internal_client/test_vault_client.py:
```diff
@@ -16,6 +16,7 @@
 # under the License.
 from __future__ import annotations
 
+import os
```
```suggestion
```

Taragolis reviewed (commented):
In test we should avoid any side effects between the different test cases, one of them create environment variables by direct assign to the `os.environ`

Environment variables for test cases should be set by `mockepatch`, `pytest.MonkeyPatch.context()` or `unittest.mock.patch.dict("os.environ", ...)`

hussein-awala reviewed (approved):
I think the idea was to load it explicitly from env var, but loading `VAULT_TOKEN` from the env var in the client is a good idea.

Just apply @Taragolis suggestions, then LGTM

V0lantis committed:
avoid any side effects between the different test cases by creating environment variables

In test we should avoid any side effects between the different test cases, one of them create environment variables by direct assign to the os.environ

Environment variables for test cases should be set by mockepatch, pytest.MonkeyPatch.context() or unittest.mock.patch.dict("os.environ", ...)

Co-authored-by: Andrey Anshin <Andrey.Anshin@taragol.is>
